File: node/src/rbac/permissions.js
'use strict';

const ROLE_PERMISSIONS = {
  Admin: {
    chat: true,
    history: true,
    metrics: true,
    kb_visibility: ['public', 'employee', 'admin'],
  },
  Employee: {
    chat: true,
    history: true,
    metrics: false,
    kb_visibility: ['public', 'employee'],
  },
  Guest: {
    chat: true,
    history: false,
    metrics: false,
    kb_visibility: ['public'],
  },
};

function getPermissions(role) {
  return ROLE_PERMISSIONS[role] || null;
}

function canAccessMetrics(role) {
  const perms = getPermissions(role);
  return Boolean(perms && perms.metrics);
}

function canAccessHistory(role) {
  const perms = getPermissions(role);
  return Boolean(perms && perms.history);
}

function allowedKbVisibility(role) {
  const perms = getPermissions(role);
  return perms ? perms.kb_visibility : [];
}

function canAccessDoc(role, visibility) {
  return allowedKbVisibility(role).includes(visibility);
}

module.exports = {
  ROLE_PERMISSIONS,
  getPermissions,
  canAccessMetrics,
  canAccessHistory,
  allowedKbVisibility,
  canAccessDoc,
};

File: python/app/rbac/permissions.py
from __future__ import annotations

ROLE_PERMISSIONS = {
    "Admin": {
        "chat": True,
        "history": True,
        "metrics": True,
        "kb_visibility": ["public", "employee", "admin"],
    },
    "Employee": {
        "chat": True,
        "history": True,
        "metrics": False,
        "kb_visibility": ["public", "employee"],
    },
    "Guest": {
        "chat": True,
        "history": False,
        "metrics": False,
        "kb_visibility": ["public"],
    },
}


def get_permissions(role: str | None):
    return ROLE_PERMISSIONS.get(role)


def can_access_metrics(role: str | None) -> bool:
    perms = get_permissions(role)
    return bool(perms and perms.get("metrics"))


def can_access_history(role: str | None) -> bool:
    perms = get_permissions(role)
    return bool(perms and perms.get("history"))


def allowed_kb_visibility(role: str | None) -> list[str]:
    perms = get_permissions(role)
    return list(perms["kb_visibility"]) if perms else []
